subprojects/core/src/main/java/org/gradle/api/internal/cache/DefaultCacheConfigurations.java

```diff
@@ -93,6 +111,21 @@ public CacheResourceConfiguration getCreatedResources() {
         return createdResourcesConfiguration;
     }
 
+    @Override
+    public void setCreatedResources(CacheResourceConfiguration createdResources) {
+        this.createdResourcesConfiguration = createdResources;
+    }
+
+    @Override
+    public Property<Cleanup> getCleanup() {
+        return cleanup;
+    }
+
+    @Override
+    public void setCleanup(Property<Cleanup> cleanup) {
```

Comment:
This is for deserializing from the configuration cache.  We _could_ deserialize, call `set()` and we would have a property backed by a property, but that wouldn't be replicating what was initially stored.  We need the setters because this is a service that we inject into the `Settings` object, rather than something that could just be deserialized along with `Settings`.  These are only on the internal interface (not the API) so I think this is ok.